cross_negative_script/cross_negative_script.gs

```diff
@@ -0,0 +1,216 @@
+  }
+  if (ADD_AD_GROUP_LEVEL_NEGATIVES) {
+    log("Adding ad group-level negatives:".toUpperCase());
+    log(" ");
+    addAdGroupLevelNegatives();
+  }
+
+  log("Script processing complete. Have a nice day!");
+}
+
+/**
+ * Adds negative keywords at the campaign level.
+ */
+function addCampaignLevelNegatives() {
+  const campaigns = [];
+  const campaignIterator = AdsApp.campaigns().withCondition("Status = ENABLED").get();
+  while (campaignIterator.hasNext()) {
+    const campaign = campaignIterator.next();
+    if (shouldProcessCampaign(campaign.getName())) {
+      const keywords = getKeywords(campaign);
+      campaigns.push({ campaign, keywords });
+    }
+  }
+  campaigns.forEach(({ campaign, keywords }) => {
+    log("Adding " + keywords.length + " keywords from " + campaign.getName() + " to all of the other campaigns.");
+    log("");
+    campaigns.forEach(({ campaign: otherCampaign }) => {
+      if (campaign.getId() !== otherCampaign.getId()) {
+        log("  - " + otherCampaign.getName());
+        addNegativesToCampaign(otherCampaign, keywords);
+      }
+    });
+    log(" ");
+    log("Adding complete.");
+    log(" ");
+  });
+}
+
+/**
+ * Adds negative keywords at the ad group level.
+ */
+function addAdGroupLevelNegatives() {
+  const campaignIterator = AdsApp.campaigns().withCondition("Status = ENABLED").get();
+  while (campaignIterator.hasNext()) {
+    const campaign = campaignIterator.next();
+    if (shouldProcessCampaign(campaign.getName())) {
+      log("Now processing the " + campaign.getName() + " campaign.");
+      log(" ");
+      const adGroups = [];
+      const adGroupIterator = campaign.adGroups().withCondition("Status = ENABLED").get();
+      while (adGroupIterator.hasNext()) {
+        const adGroup = adGroupIterator.next();
+        const keywords = getKeywords(adGroup);
+        adGroups.push({ adGroup, keywords });
+      }
+      adGroups.forEach(({ adGroup, keywords }) => {
+        if (keywords.length > 0) {
+          log(
+            "Adding " +
+              keywords.length +
+              " keywords from the " +
+              adGroup.getName() +
+              " ad group to the other ad groups in the " +
+              campaign.getName() +
+              " campaign."
+          );
+          log("");
+          adGroups.forEach(({ adGroup: otherAdGroup }) => {
+            if (adGroup.getId() !== otherAdGroup.getId()) {
+              addNegativesToAdGroup(otherAdGroup, keywords);
+            }
+          });
+          log("Adding complete.");
+          log(" ");
+        }
+      });
+    }
+  }
+}
+
+/**
+ * Determines if a campaign should be processed based on its name.
+ * @param {string} campaignName - The name of the campaign.
+ * @returns {boolean} - True if the campaign should be processed, false otherwise.
+ */
+function shouldProcessCampaign(campaignName) {
+  const includeRegex = INCLUDE_CAMPAIGN_NAMES.length > 0 ? new RegExp(INCLUDE_CAMPAIGN_NAMES.join("|"), "i") : null;
```

Comment:
⚠️ If you're doing a case-insensitive match, I'd make it clear your user inputs at the start ignore case. Might end up with some unexpected inclusions/exclusions otherwise.